### PR TITLE
Fix bugs when enable e2e diagnostic with HTTP protocol

### PR DESCRIPTION
**Pull request by SLdragon**
# Checklist
- [ ] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-c/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [ ] This pull-request is submitted against the `master` branch. 

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

# Description of the solution

### CODE REVIEW

**Comment by erich-wang** [comment]
@CIPop the issue is that we will fail to send telemetry message if setting **DeviceClient.DiagnosticSamplingPercentage = 100** with Http protocol.
the following code snippet describes the issue
``` csharp
            var client = DeviceClient.CreateFromConnectionString(deviceConnectionString, TransportType.Http1);
            client.DiagnosticSamplingPercentage = 100;
            var message = new Message(Encoding.ASCII.GetBytes("hello"));
            client.SendEventAsync(message).Wait(); //get exception here
```
The root cause is diagnostic related system properties are added into message but underlying http protocol handler could not recognize these properties.
The fix is to not allow set DiagnosticSamplingPercentage for Http protocol to align with e2e diagnostic design.

File: device/Microsoft.Azure.Devices.Client/DeviceClient.cs [hunk 1]
```diff
@@ -1463,6 +1469,21 @@ internal void OnReportedStatePatchReceived(TwinCollection patch)
                 await ApplyTimeout(operationTimeoutCancellationToken => this.InnerHandler.DisableMethodsAsync(operationTimeoutCancellationToken)).ConfigureAwait(false);
             }
         }
+
+        private bool IsE2EDiagnosticSupportedProtocol()
+        {
+            var transportSetting = this.transportSettings.FirstOrDefault();
```
Comment (CIPop) [inline device/Microsoft.Azure.Devices.Client/DeviceClient.cs]: This is looking at only the first item in the list - what about the fallback case where multiple protocol configurations are specified like {Amqp, Mqtt, Http}. If for some reason both Amqp and Mqtt fail, the code will fail when Diagnostics are attempted with Http.

File: device/tests/Microsoft.Azure.Devices.Client.Test/DeviceClientTests.cs [hunk 2]
```diff
@@ -75,6 +75,22 @@ public void DeviceClient_SetDiagnosticSamplingPercentageOutOfRange_Fail()
             }
         }
 
+        [TestMethod]
+        [TestCategory("IoTHubClientDiagnostic")]
+        public void DeviceClient_StartDiagLocallyThatDoNotSupport_ThrowException()
+        {
+            DeviceClient deviceClient = DeviceClient.CreateFromConnectionString(fakeConnectionString, TransportType.Http1);
```
Comment (CIPop) [inline device/tests/Microsoft.Azure.Devices.Client.Test/DeviceClientTests.cs]: Please add another test where multiple transport types are used: transport choices w/o Http1 should work while those containing Http1 anywhere in the list should fail.

File: device/Microsoft.Azure.Devices.Client/DeviceClient.cs (hunk 1 above)
Comment (SLdragon) [reply device/Microsoft.Azure.Devices.Client/DeviceClient.cs]: Thanks for your suggestions, we are missing that case, we will fix it.

File: device/tests/Microsoft.Azure.Devices.Client.Test/DeviceClientTests.cs (hunk 2 above)
Comment (SLdragon) [reply device/tests/Microsoft.Azure.Devices.Client.Test/DeviceClientTests.cs]: fixed

**Pull request merged**